core/src/main/scala/akka/persistence/dynamodb/internal/MonotonicTimestamps.scala

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (C) 2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.dynamodb.internal
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.actor.typed.SupervisorStrategy
+import akka.actor.typed.scaladsl.AskPattern.Askable
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.TimerScheduler
+import akka.annotation.InternalApi
+import akka.persistence.Persistence
+
+import scala.annotation.tailrec
+import scala.collection.immutable.SortedSet
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.jdk.CollectionConverters.IteratorHasAsScala
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.time.Instant
+import java.time.{ Duration => JDuration }
+import java.util.concurrent.ConcurrentHashMap
+import java.time.temporal.ChronoUnit
+
+object MonotonicTimestamps extends ExtensionId[MonotonicTimestamps] {
```

Comment:
May just be me, but the implementation looks more complex than what we need. Feel that it could be simpler, but will think through it some more too.

The journal is an actor, so could also track directly. Also see how `writesInProgress` are tracked in the journal implementation.

Reply:
Haven't looked at the details here. Would it be enough to increase the time in InstantFactory when detecting clock skew? At the point the warning is logged in https://github.com/akka/akka-persistence-dynamodb/pull/110

Reply:
Yeah, was thinking the same. We could just bump in InstantFactory. That will then be monotonically increasing micros until the current time catches up and it reverts to regular timestamps again. And probably check against a configurable tolerance setting, so it can only be skewed by so much, otherwise error.